### DISCUSSION TITLE
how to change this torch to flax?

### DISCUSSION THREAD

**Discussion by lu-ming-lei** [google/flax#1385] [Q&A]
optimizer = torch.optim.Adam([variable1, variable2], 0.001)

how to change this torch to flax?

I try this but no success:  optimizer = flax.optim.Adam([variable1, variable2], 0.001)

**Comment by joaogui1** [comment]
Hey @lu-ming-lei can you provide a longer code snippet? In Flax we don't pass the variables we want to optmize when initialising the optimizer

**Comment by lu-ming-lei** [comment]
optimizer = torch.optim.Adam([
  {'params': variables1, 'lr':0.1},  
   {'params': variables2, 'lr':0.01}, 
  ])
how to change this code into flax?

**Comment by jheek** [comment]
We currently recommend new users to use [optax](https://github.com/deepmind/optax) for optimizing parameters. Please take a look at their documentation and the examples in the READMe for how it is used. 

Generally JAX uses functional APIs (so no mutable variables). Therefore, there is not an obvious 1:1 mapping from torch to JAX because torch heavily uses variables which JAX does not have.
